vpc/service/assign_addresses_v3.go

```diff
@@ -1623,3 +1623,53 @@ func assignSpecificIPv4AddressV3(ctx context.Context, tx *sql.Tx, branchENI *ec2
 		PrefixLength: uint32(prefixlength),
 	}, nil
 }
+
+func assignNextIPv6Address(ctx context.Context, tx *sql.Tx, subnetid string) (net.IP, error) {
+	sequenceName := fmt.Sprintf("v6seq_%s", strings.ReplaceAll(subnetid, "-", "_"))
+	_, err := tx.ExecContext(ctx, fmt.Sprintf("CREATE SEQUENCE IF NOT EXISTS %s MINVALUE 0 MAXVALUE 65534 CACHE 10 CYCLE", sequenceName))
+	if err != nil {
+		return nil, fmt.Errorf("Cannot create sequence %s: %w", sequenceName, err)
+	}
+
+	/*
+			The address layout looks like:
+
+		 ┌───────────────────────────┬──────────────┬───────────────────────────┬────────────────────┐
+		 │ AWS Subnet CIDR (64-bits) │ 0s (16 bits) │ Time in Seconds (32-bits) │ Sequence (16-bits) │
+		 └───────────────────────────┴──────────────┴───────────────────────────┴────────────────────┘
+
+	*/
+
+	row := tx.QueryRowContext(ctx, "SELECT nextval($1)", sequenceName)
```

Comment:
Why do the unix time at all if we already are making the DB do the work of accounting for the sequence?

Reply:
Because you're not guaranteed to have this number always increment, and you generally want to do what you can to spread it out. For example:

```sql
START TRANSACTION;
SELECT nextval('foo');
ROLLBACK;
START TRANSACTION;
SELECT nextval('foo'); -- It will return the same value
```